qcodes/utils/logger.py

```diff
@@ -0,0 +1,439 @@
+import io
+import logging
+
+import os
+from pathlib import Path
+from collections import OrderedDict
+from contextlib import contextmanager
+from copy import copy
+
+from typing import Optional, List, Union, Sequence
+
+import pandas
```

Comment:
`pandas` is the only non std and non qcodes dependency in this module. Moving these functions to a new module would require exactly the same dependencies in the new module.

Reply:
Correct. and it will release the `qcodes.utils.logger` from requiring `pandas`. 

I'm thinking of `qcodes.utils.logger` module without `pandas` and smth like `qcodes.utils.logger_pandas_analysis` that does depend on `pandas`. This way the general logging features can be imported without having pandas installed. What do you think? Or is it unimportant and too much fuzz for a small benefit?

Reply:
I think we agreed on making it a dependency: https://github.com/QCoDeS/Qcodes/pull/1309#discussion_r224741658

Reply:
ok, sure.